### PR TITLE
Promote did_you_mean to default gem

### PR DESCRIPTION
**Pull request by kddeisz**
At the moment, there are some problems with regard to bundler + did_you_mean because of did_you_mean being a bundled gem. Since the vendored version of thor inside bundler and ruby itself explicitly requires did_you_mean, it can become difficult to load it when using Bundler.setup. See this issue: https://github.com/yuki24/did_you_mean/issues/117#issuecomment-482733159 for more details.

This is my first PR to Ruby so apologies if I didn't follow the directions correctly. As for the PR itself, I basically just updated `tool/sync_default_gems.rb`, made a couple of doc updates, and then ran the script for just `did_you_mean`. 

@yuki24 feel free to open your own PR or whatever you need to do, this is just to get it started.

### CODE REVIEW

**Comment by yuki24** [comment]
Thanks for taking the time to send this PR! I will look at it more closely later. I might have to replace `minitest` with `test-unit` but that shouldn’t be too difficult. I’ll also take a look at the test failures to see if I can fix them.

**Comment by yuki24** [comment]
The repo has been moved to [ruby/did_you_mean](https://github.com/ruby/did_you_mean). The next step would be to replace `minitest` with `test-unit`. I will try to do that this weekend.

**Comment by kddeisz** [comment]
Nice - I've updated this PR to reflect the new location

**Comment by kddeisz** [comment]
Some tests failing that I think https://github.com/ruby/did_you_mean/pull/131 will fix.

**Comment by kddeisz** [comment]
Okay good it's to the point of just breaking other tests now, which can probably be relaxed. Looks like they're all with TracePoint and throwing exceptions and matching on errors. As in:

https://github.com/ruby/ruby/commit/20a8cd310dfe847cff5c50f184626be46b1c7599/checks#step:10:153

**Comment by yuki24** [comment]
Alright, I have removed experimental features that depended on `TracePoint`. I don't think it is worth maintaining and it's more important to bring it in. Let's see how it goes.

File: tool/sync_default_gems.rb [hunk 1]
```diff
@@ -210,6 +212,11 @@ def sync_default_gems(gem)
     rm_rf("test/racc/lib")
     rm_rf("lib/racc/cparse-jruby.jar")
     `git checkout ext/racc/cparse/README ext/racc/cparse/depend`
+  when "did_you_mean"
+    rm_rf(%w[lib/did_you_mean* test/did_you_mean])
+    cp_r(Dir.glob("#{upstream}/lib/did_you_mean*"), "lib")
+    cp_r("#{upstream}/did_you_mean.gemspec", "lib/did_you_mean")
+    cp_r("#{upstream}/test", "test/did_you_mean")
```
Comment (yuki24) [inline tool/sync_default_gems.rb]: Is it possible to remove this file `test/did_you_mean/tree_spell/test_explore.rb`? I know it's awkward, but this test particularly takes more than 5min and is not written in a normal testing fashion. I'm looking to improve it but haven't had time to do so. I would like to exclude it until we come up with a better solution.

```suggestion
    cp_r("#{upstream}/test", "test/did_you_mean")
    rm_rf(%w[test/did_you_mean/tree_spell/test_explore.rb])
```

**Comment by yuki24** [comment]
closing in favor of https://github.com/ruby/ruby/pull/2689.

**Pull request closed without merging**